### DISCUSSION TITLE
I've a question, despite having cognitive impairments, how am I able to memorize programming languages such as C++, Python, and x86 assembly, especially when I can recall the specific code for 'Hello, World!' in both assembly and C++?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#141728] [Programming Help]
### Body

I've a question, despite having cognitive impairments, how am I able to memorize programming languages such as C++, Python, and x86 assembly, especially when I can recall the specific code for 'Hello, World!' in both assembly and C++?

Python:
```python
print("Hello, World!")
```

C++:
```cpp
#Include <iostream>
int main(){
    std::cout << "Hello, World!";
    return 0;
}
```

```assembly
SECTION .data
     msg: db "Hello, World!"
SECTION .text
    GLOBAL _start

_start:
   MOV EAX, 4
   MOV EBX, 1
   MOV ECX, msg
  MOV EDX, 13
  INT 0x80
  MOV EAX, 1
  XOR EBX, EBX
  INT 0x80

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Zuba43** [comment]
It's all about understanding the syntax of the language of your choice. 
The program flow and then a bit of advanced features like OOP Features

**Reply by ghost** [reply]
Should I not use C++ if I have cognitive problems? I think I should switch to Python again
